src/main/java/de/tum/in/www1/artemis/domain/exam/event/ExamAttendanceCheckEvent.java

```diff
@@ -29,9 +29,6 @@ public void setTextContent(String textContent) {
 
     @Override
     public ExamAttendanceCheckEventDTO asDTO() {
-        var dto = new ExamAttendanceCheckEventDTO();
-        super.populateDTO(dto);
-        dto.setText(textContent);
-        return dto;
+        return new ExamAttendanceCheckEventDTO(this.getId(), this.getCreatedBy(), this.getCreatedDate(), textContent);
```

Comment:
IMO it would make more sense to add an `public static of(ExamAttendenceCheck e)` method to the `ExamAttendenceCheckDTO`. We have these methods in multiple DTOs already and it would make the code more consistent.

Reply:
Additionally to the consistency point of Raphael, it makes sense to move this convertion method into the DTO as several DTOs for one class may exists. Therefore the concrete mapping should be handled by the individual DTO to separate concerns.

Reply:
This currently uses inheritance and polymorphism, since asDTO() is called on objects that have the abstract super class as static type. This will be solved in a follow up